app/scripts/controllers/transactions/index.js

```diff
@@ -2081,19 +2083,47 @@ export default class TransactionController extends EventEmitter {
       eip1559Version = eip1559V2Enabled ? '2' : '1';
     }
 
+    const contractInteractionTypes = [
+      TRANSACTION_TYPES.CONTRACT_INTERACTION,
+      TRANSACTION_TYPES.TOKEN_METHOD_APPROVE,
+      TRANSACTION_TYPES.TOKEN_METHOD_SAFE_TRANSFER_FROM,
+      TRANSACTION_TYPES.TOKEN_METHOD_SET_APPROVAL_FOR_ALL,
+      TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER,
+      TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER_FROM,
+      TRANSACTION_TYPES.SMART,
+      TRANSACTION_TYPES.SWAP,
+      TRANSACTION_TYPES.SWAP_APPROVAL,
+    ].includes(type);
+
+    let transactionType;
+    if (type === TRANSACTION_TYPES.CANCEL) {
+      transactionType = TRANSACTION_TYPES.CANCEL;
+    } else if (type === TRANSACTION_TYPES.SIMPLE_SEND) {
+      transactionType = TRANSACTION_TYPES.SIMPLE_SEND;
+    } else if (type === TRANSACTION_TYPES.DEPLOY_CONTRACT) {
+      transactionType = TRANSACTION_TYPES.DEPLOY_CONTRACT;
+    } else if (contractInteractionTypes) {
+      transactionType = TRANSACTION_TYPES.CONTRACT_INTERACTION;
+    }
```

Comment:
```suggestion
    let transactionType = TRANSACTION_TYPES.SIMPLE_SEND;
    if (type === TRANSACTION_TYPES.CANCEL) {
      transactionType = TRANSACTION_TYPES.CANCEL;
    } else if (type === TRANSACTION_TYPES.DEPLOY_CONTRACT) {
      transactionType = TRANSACTION_TYPES.DEPLOY_CONTRACT;
    } else if (contractInteractionTypes) {
      transactionType = TRANSACTION_TYPES.CONTRACT_INTERACTION;
    }
```


This prevents us from having potentially undefined types, however with the way the code currently works and without the originalType yet being recorded this will result in retry being labeld as "Simple Send" I suggest that we first keep RETRY as a type then remove it in the PR that is handling original type.

Reply:
@brad-decker So you are saying that we first keep RETRY as a `transaction_type` in this PR and than when this get merged to remove it in the PR that is handling original type? I need to add new if case:

>  if (type === TRANSACTION_TYPES.RETRY) {
>     transactionType = TRANSACTION_TYPES.RETRY;
> }

Reply:
@brad-decker [Here](https://github.com/MetaMask/metamask-extension/pull/15175/commits/ebeb2392fc54ee7274ba0bb161473f067be67d33) I made this requested changes.

Reply:
We should also perserve the retry type until the retry work lands.

Reply:
Let's have RETRY have transaction_type = retry and ask @adnansahovic to include the respective change to his PR. cc: @danjm